src/Symfony/EventListener/DeserializeListener.php

```diff
@@ -72,11 +77,28 @@ public function onKernelRequest(RequestEvent $event): void
         if (null !== $data) {
             $context[AbstractNormalizer::OBJECT_TO_POPULATE] = $data;
         }
-
-        $request->attributes->set(
-            'data',
-            $this->serializer->deserialize($request->getContent(), $context['resource_class'], $format, $context)
-        );
+        try {
+            $request->attributes->set(
+                'data',
+                $this->serializer->deserialize($request->getContent(), $context['resource_class'], $format, $context)
+            );
+        } catch (PartialDenormalizationException $e) {
+            $violations = new ConstraintViolationList();
+            foreach ($e->getErrors() as $exception) {
+                if (!$exception instanceof NotNormalizableValueException) {
+                    continue;
+                }
+                $message = sprintf('The type of the "%s" attribute must be "%s", "%s" given.', $exception->getPath(), implode(', ', $exception->getExpectedTypes() ?? []), $exception->getCurrentType());
```

Comment:
Maybe it would be better to use by default a message like in Symfony Validator componen: `This value should be of type {{ type }}.` so it would be translated if validator and translation components are dependencies of application?